geomesa-arrow/geomesa-arrow-datastore/src/main/scala/org/locationtech/geomesa/arrow/data/ArrowDataStore.scala

```diff
@@ -0,0 +1,103 @@
+/***********************************************************************
+* Copyright (c) 2013-2017 Commonwealth Computer Research, Inc.
+* All rights reserved. This program and the accompanying materials
+* are made available under the terms of the Apache License, Version 2.0
+* which accompanies this distribution and is available at
+* http://www.opensource.org/licenses/apache2.0.php.
+*************************************************************************/
+
+package org.locationtech.geomesa.arrow.data
+
+import java.io.{FileOutputStream, IOException, InputStream, OutputStream}
+import java.net.URL
+
+import org.geotools.data._
+import org.geotools.data.simple.SimpleFeatureSource
+import org.geotools.data.store.{ContentDataStore, ContentEntry, ContentFeatureSource}
+import org.geotools.feature.NameImpl
+import org.locationtech.geomesa.arrow.io.{SimpleFeatureArrowFileReader, SimpleFeatureArrowFileWriter}
+import org.locationtech.geomesa.index.metadata.{GeoMesaMetadata, HasGeoMesaMetadata}
+import org.locationtech.geomesa.index.stats.{GeoMesaStats, HasGeoMesaStats, NoOpMetadata, UnoptimizedRunnableStats}
+import org.locationtech.geomesa.utils.io.WithClose
+import org.opengis.feature.`type`.Name
+import org.opengis.feature.simple.{SimpleFeature, SimpleFeatureType}
+import org.opengis.filter.Filter
+
+import scala.util.Try
+import scala.util.control.NonFatal
+
+class ArrowDataStore(val url: URL) extends ContentDataStore with FileDataStore
+    with HasGeoMesaMetadata[String] with HasGeoMesaStats {
+
+  import org.locationtech.geomesa.arrow.allocator
```

Comment:
I think this is actually used implicitly when creating the writers/readers, like on line 53